Task_DEV_4/Task_DEV_4/Parallelogram.cs

```diff
@@ -2,37 +2,57 @@
 
 namespace Task_DEV_4
 {
-    class Parallelogram : Square
+    class Parallelogram : Figure
     {
-        public double HeightA { get; set; }
-        public double Angle { get; set; }
-        public double SideB { get; set; }
+        private const string ERROR_MESSAGE = "Side or angle cannot be less equals than 0.";
 
-        public Parallelogram(double sideA, double heightA) : base(sideA)
-        {
-            HeightA = heightA;
-        }
+        public double Angle { get; set; }
+        public double FirstSide { get; set; }
+        public double SecondSide { get; set; }
 
-        public Parallelogram(double sideA, double angle, double sideB) : base(sideA)
+        public Parallelogram(string color, string name, double secondSide, double angle, double firstSide) : base(color, name)
         {
-            SideB = sideB;
+            SecondSide = secondSide;
+            FirstSide = firstSide;
             Angle = angle;
+            CheckForExistance();
         }
 
         public override double CalculateArea()
         {
-            if (HeightA > 0 && SideA > 0)
+            if (FirstSide > 0 && SecondSide > 0 && Angle > 0)
```

Comment:
Why are you using this check in this method?

Reply:
Done.